resources/qml/Preferences/Materials/MaterialsList.qml

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) 2018 Ultimaker B.V.
```

Comment:
It would be nice if we could split up this huge qml file into a few more simple components.

Reply:
It could be. I originally tried to just make a single cascading tree component but Qt did not like my recursive antics so I settled for just splitting the "material slot" piece out. I could split out the fold-out headers too though. Will do that tomorrow.

Reply:
Done.

Reply:
Also fixed that unfavoriting bug you found. 😉